### PR TITLE
Retry context diagram endpoints

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_api_client.py [hunk 1]
```diff
@@ -103,6 +103,7 @@ class APIClient:
                     "workflows/.*",
                     "hostedextractors/.*",
                     "postgresgateway/.*",
+                    "context/diagram/*",
```
Comment (haakonvt) [inline cognite/client/_api_client.py:106]: Missing a dot I believe

**Review by haakonvt — COMMENTED** [review]
Needs a test 👌

File: cognite/client/_api_client.py (hunk 1 above)
Comment (doctrino) [inline cognite/client/_api_client.py:106]:
```suggestion
                    "context/diagram/.*",
```

File: tests/tests_unit/test_api_client.py [hunk 2]
```diff
@@ -1361,6 +1361,11 @@ def test_is_retryable_resource_api_endpoints(self, api_client_with_token, method
                 ("POST", "https://api.cognitedata.com/api/v1/projects/bla/raw/dbs/db/tables/t", False),
                 ("POST", "https://api.cognitedata.com/api/v1/projects/bla/raw/dbs/db/tables/t/rows", True),
                 ("POST", "https://api.cognitedata.com/api/v1/projects/bla/raw/dbs/db/tables/t/rows/delete", True),
+                # Engineering diagrams
+                ("POST", "https://api.cognitedata.com/api/v1/projects/bla/context/diagram/convert", True),
+                ("POST", "https://api.cognitedata.com/api/v1/projects/bla/context/diagram/detect", True),
+                ("GET", "https://api.cognitedata.com/api/v1/projects/bla/context/diagram/detect/123", True),
```
Comment (haakonvt) [inline tests/tests_unit/test_api_client.py:1367]:
```suggestion
                ("GET", "https://api.cognitedata.com/api/v1/projects/bla/context/diagram/convert/123", True),
```

**Review by haakonvt — APPROVED** [review]

**Pull request merged**